ksqldb-parser/src/test/java/io/confluent/ksql/parser/KsqlParserTest.java

```diff
@@ -938,6 +939,37 @@ public void shouldSetWithinExpressionWithSingleWithin() {
     assertThat(withinExpression.getBefore(), is(10L));
     assertThat(withinExpression.getAfter(), is(10L));
     assertThat(withinExpression.getBeforeTimeUnit(), is(TimeUnit.SECONDS));
+    assertThat(withinExpression.getGrace(), is(Optional.empty()));
+    assertThat(join.getType(), is(JoinedSource.Type.INNER));
+  }
+
+  @Test
+  public void shouldSetWithinExpressionWithSingleWithinAndGracePeriod() {
```

Comment:
To follow up to @vcrfxia comment -- should we add a test that verifies that we fail if `GRACE` is specified but if the feature flag disables the fix? (not sure if a unit test or QTT test should cover it?)

Reply:
I'll add a test in a follow-up PR